### ISSUE TITLE
Mapping value to text does't work in singlestat panel

### ISSUE THREAD

**Issue by romanponomarev** [alexanderzobnin/grafana-zabbix#138]
Hello, Alexander.
Mapping value to text does't work in singlestat panel. I don't understend if it's the problem of grafana or grafana-zabbix.

Installed APP:
zabbix - 2.4.7
grafana - 2.6
grafana-zabbix - 2.5.1

I created singlestat panel, added query "IT Service" and chose IT service and IT service property - Status...
![pic1](https://cloud.githubusercontent.com/assets/15785694/12149929/b9ce3e4a-b4d0-11e5-802f-cc04c0fb2a34.PNG)

The current status value is 2.
Responce:
{"jsonrpc":"2.0","result":{"1":{"status":"2","problems":[],"sla":[{"from":1352452201,"to":1353057001,"sla":100,"okTime":604800,"problemTime":0,"downtimeTime":0}]}},"id":1}

![pic2](https://cloud.githubusercontent.com/assets/15785694/12149950/d4a0d8a4-b4d0-11e5-9e42-96f474407030.PNG)
but while mapping of value to text,  the current status value is 0, for some reason.
![pic3](https://cloud.githubusercontent.com/assets/15785694/12149956/db70faf6-b4d0-11e5-9a94-a817e6cf26b0.PNG)
Could you help me with this problem ?

**Comment by alexanderzobnin** [comment]
@romanponomarev Looks strange. It's may be Grafana bug, because my plugin only provide raw value to the panel.

**Comment by romanponomarev** [comment]
Thank you for the answer.
I created the [issue](https://github.com/grafana/grafana/issues/3667) in grafana project.

**Comment by romanponomarev** [comment]
@alexanderzobnin I've found the decision of the problem: the plugin returns multiple value to the panel.
The panel shows the correct value, but the mapping gives the last number from the array (it is 0, see Responce).
The panel works correctly and plugin too. What must process this exception: grafana or grafana-zabbix? I need your help. :)

**Comment by alexanderzobnin** [comment]
Zabbix API response is handled by `handleSLAResponse` function:
```
this.handleSLAResponse = function (itservice, slaProperty, slaObject) {
      var targetSLA = slaObject[itservice.serviceid].sla[0];
      if (slaProperty.property === 'status') {
        var targetStatus = slaObject[itservice.serviceid].status;
        return {
          target: itservice.name + ' ' + slaProperty.name,
          datapoints: [
            [targetStatus, targetSLA.to * 1000]
          ]
        };
...
```
And this function returns array with single element:
```
datapoints: [
            [targetStatus, targetSLA.to * 1000]
          ]
```

You can test data returned to panel. Add logging to end of `query()` method in `datasource.js` (line 250):
```
return $q.all(_.flatten(promises)).then(function (results) {
        var timeseries_data = _.flatten(results);
        
        // Returned data logging
        console.log(timeseries_data);
        return { data: timeseries_data };
      });
```

Looks realy strange - I can't reproduce this bug in my environment.

**Comment by wmbzz** [comment]
I think I'm having the same issue. I have a item in Zabbix monitoring a Windows service, when it's running the value is 0 and when it's stopped the value is 6. I created a singlestat panel pointing to this trigger and stopped the service. Zabbix caught it and even sent me the e-mail, but the panel doesn't update, it keeps showing 0. How can I troubleshoot this to help you?

**Comment by alexanderzobnin** [comment]
@wmbzz Can you attach screenshots of your panel's config (Metrics and Options tabs)?

**Comment by romanponomarev** [comment]
@alexanderzobnin  I added, but I don't know how to see the log from the console.
![pic6](https://cloud.githubusercontent.com/assets/15785694/12217362/08e5ac1c-b728-11e5-97d6-9629016251e1.PNG)

Here are the data of the browser debug mod:
![pic4](https://cloud.githubusercontent.com/assets/15785694/12217367/3ebae1c2-b728-11e5-86f2-a0d9a07fcd38.PNG)

![pic5](https://cloud.githubusercontent.com/assets/15785694/12217369/4649b7f6-b728-11e5-9c7e-e146488abe5b.PNG)

What data I must give to analyse the problem ?

**Comment by alexanderzobnin** [comment]
Hm... Data returned by plugin is right:
```
datapoints: [
  [2, 145...]
]
```

**Comment by romanponomarev** [comment]
Maybe the matter is in 000 after 145...
Number in network tab in debugger without ...000

**Comment by alexanderzobnin** [comment]
No, it's just converted seconds to ms.

**Comment by romanponomarev** [comment]
Good day, @alexanderzobnin.
I've found the reason of my problem:
Zabbix returns the value "status" as integer, but zabbix API returns "status" as string ([see more for details](https://www.zabbix.com/documentation/2.4/manual/api/reference/service/getsla?s[]=getsla));
Grafana-zabbix returns the value to Grafana as is (string);
![pic7](https://cloud.githubusercontent.com/assets/15785694/12261340/a0af1848-b94a-11e5-898e-9cef75ce6053.PNG)

Grafana binds 0 to all string values:
![pic8](https://cloud.githubusercontent.com/assets/15785694/12261346/b703c8e6-b94a-11e5-81b3-ec838e9369b4.PNG)

Alexander, can you help me to correct this problem ?

**Comment by alexanderzobnin** [comment]
@romanponomarev yes, I will fix it.

**Comment by romanponomarev** [comment]
Thank you!

**Comment by alexanderzobnin** [comment]
@romanponomarev This fix now in branch for grafana 3.0 will be included in next plugin version.
